pkg/vex/sbomref.go

```diff
@@ -0,0 +1,113 @@
+package vex
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/fanal/artifact"
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/sbom/core"
+	"github.com/aquasecurity/trivy/pkg/types"
+)
+
+type SBOMReferenceSet struct {
+	Vexes []VEX
+}
+
+func NewSBOMReferenceSet(report *types.Report) (*SBOMReferenceSet, error) {
+
+	if report.ArtifactType != artifact.TypeCycloneDX {
+		return nil, xerrors.Errorf("externalReferences can only be used when scanning CycloneDX SBOMs: %w", report.ArtifactType)
+	}
+
+	var externalRefs = report.BOM.ExternalReferences()
+	urls := parseToURLs(externalRefs)
+
+	v, err := retrieveExternalVEXDocuments(urls, report)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to fetch external VEX document: %w", err)
+	}
```

Comment:
If no valid VEX documents are found, `v` will be nil, and in that case, it seems acceptable to also set SBOMReferenceSet to nil and skip it. I wasn't sure whether a nil check was needed for the return value of `NewSBOMReferenceSet()`, but after making this change, it is necessary.

```suggestion
	v, err := retrieveExternalVEXDocuments(urls, report)
	if err != nil {
		return nil, xerrors.Errorf("failed to fetch external VEX document: %w", err)
	} else if v == nil {
		return nil, nil
	}
```